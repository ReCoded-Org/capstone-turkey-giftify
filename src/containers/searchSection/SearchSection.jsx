import React, { useState } from 'react';
import Boards from '../../components/boards/Boards';
import "./SearchSection.css";

function SearchSection () {

    const [selectedItem, setSelectedItem] = useState([])

    // const [selectedCategory, setSelectedCategory] = useState([])
    console.log(selectedItem)



    const [input, setInput] = useState ("");
    const [output, setOutput] = useState ([]);

    function handleChange(event){
        setInput(event.target.value)
    }

    function handleSubmit(event) {
        event.preventDefault()
        const searchResult = selectedItem
        .filter((val) => {
              if(input === ""){
                return (val)
              } else if (val.item.toLowerCase().includes(input.toLowerCase())){
                return (val)
              } else if (val.type.toLowerCase().includes(input.toLowerCase())){
                return (val)
              }
            })
            .map((item) => { 
              return(
                <div key = {item.id}className = "oneCard"> 
                  <h3>{item.item}</h3> 
                  <p>Use Duration: {item.useDuration} </p>
                  <br/>
                  <img className = "itemImage"  src={item.imageURL} alt = "clothes" />
                </div>
                  )
            })
            setOutput(searchResult)
    }

    return(
        <div>
            <h2 className = "parag">Or would you like to search for yourself ?</h2>

            <form className = "form" onSubmit= {handleSubmit} >
                <label>
                    <input className = "searchBar" type="text" name="name" onChange={handleChange} value={input} />
                </label>
                <input className = "submit" type="submit" value="Submit" />
            </form>

            <div className = "cards">
            {output}
        </div>
            <div><Boards setSelectedItem = {setSelectedItem}  /></div>
        </div>
    )
}
export default SearchSection