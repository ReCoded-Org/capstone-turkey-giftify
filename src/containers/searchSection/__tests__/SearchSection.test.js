import React from "react";
import SearchSection from "./../SearchSection";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("matches snapshot", () => {
  const tree = renderer.create(<SearchSection />).toJSON();
  expect(tree).toMatchSnapshot();
});