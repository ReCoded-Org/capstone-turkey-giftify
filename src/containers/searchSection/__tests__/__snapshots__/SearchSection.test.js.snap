// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <h2
    className="parag"
  >
    Or would you like to search for yourself ?
  </h2>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <label>
      <input
        className="searchBar"
        name="name"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <input
      className="submit"
      type="submit"
      value="Submit"
    />
  </form>
  <div
    className="cards"
  />
  <div>
    <div>
      <div
        className="categories"
      >
        <button
          className="oneCategory householdAppliances"
          onClick={[Function]}
        >
          Household Appliances
        </button>
        <button
          className="oneCategory furniture"
          onClick={[Function]}
        >
          Furniture
        </button>
        <button
          className="oneCategory clothes"
          onClick={[Function]}
        >
          Clothes
        </button>
        <button
          className="oneCategory"
          onClick={[Function]}
        >
          All
        </button>
      </div>
    </div>
  </div>
</div>
`;
